### ISSUE TITLE
`Style/KeywordArgumentsMerging` can cause an infinite autocorrection loop

### ISSUE THREAD

**Issue by kevinrobell-st** [rubocop/rubocop#14597]
When `Style/KeywordArgumentsMerging` autocorrects some code, it can lead to an infinite loop between the `Layout/ArgumentAlignment` and `Layout/HashAlignment` cops. The autocorrection still seems to succeed though which is also interesting.

Note: I've omitted some file and directory names below but the examples should be reproducible without them.

Note 2: When `EnforcedStyle: with_fixed_indentation` is not enabled for the `Layout/ArgumentAlignment` cop, no error is reported.

--------

## Steps to reproduce the problem

### Before

Example file:

```rb
# frozen_string_literal: true

# 1. Infinite loop
build(
  :factory_name,
  **options.merge(
    :first => 1,
    :second => 2
  )
)

# 2. This works fine
build(:factory_name, **options.merge(
  :first => 1,
  :second => 2
))

# 3. This works fine
build(:factory_name, **options.merge(:first => 1, :second => 2))
```

Example settings:

```yml
Layout/ArgumentAlignment:
  Enabled: true
  EnforcedStyle: with_fixed_indentation

Layout/HashAlignment:
  Enabled: true
  EnforcedColonStyle:
    - key
    - table
  EnforcedHashRocketStyle:
    - key
    - table

Style/KeywordArgumentsMerging:
  Enabled: true
```

```console
$ bundle exec rubocop spec/example.rb
Inspecting 1 file
C

Offenses:

spec/example.rb:6:5: C: [Correctable] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
  **options.merge( ...
    ^^^^^^^^^^^^^^
spec/example.rb:13:24: C: [Correctable] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
build(:factory_name, **options.merge( ...
                       ^^^^^^^^^^^^^^
spec/example.rb:19:24: C: [Correctable] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
build(:factory_name, **options.merge(:first => 1, :second => 2))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 3 offenses detected, 3 offenses autocorrectable

Tip: Based on detected gems, the following RuboCop extension libraries might be helpful:
  * rubocop-capybara (https://rubygems.org/gems/rubocop-capybara)

You can opt out of this message by adding the following to your config (see https://docs.rubocop.org/rubocop/extensions.html#extension-suggestions for more options):
  AllCops:
    SuggestExtensions: false
```

## Expected behavior
It autocorrects code without an infinite loop.

## Actual behavior

```console
$ bundle exec rubocop spec/example.rb -a --debug
For /Users/kevinrobell/Documents/source/...: configuration from ...
Inheriting configuration from ...
Default configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-1.80.2/config/default.yml
Plugin configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-performance-1.26.0/config/default.yml
Plugin configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-rails-2.33.4/config/default.yml
Plugin configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-rspec-3.7.0/config/default.yml
Plugin configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-factory_bot-2.27.1/config/default.yml
Plugin configuration from /Users/kevinrobell/.asdf/installs/ruby/3.1.6/lib/ruby/gems/3.1.0/gems/rubocop-rspec_rails-2.31.0/config/default.yml
Inheriting configuration from ...
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/kevinrobell/Documents/source/.../spec/example.rb
Loading cache from /Users/kevinrobell/.cache/rubocop_cache/b177bd2dc88d69449cfb1bb9fbf29a652b8d94c1/6d7a3b621ca1730e04accd938619e4bdab66cfb1/bfaef681b4919d15feed60f3e6b07c375dbca96d
Infinite loop detected in /Users/kevinrobell/Documents/source/.../spec/example.rb and caused by Layout/ArgumentAlignment, Layout/HashAlignment -> Layout/ArgumentAlignment, Layout/HashAlignment
C

Offenses:

spec/example.rb:6:5: C: [Corrected] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
  **options.merge( ...
    ^^^^^^^^^^^^^^
spec/example.rb:7:5: C: [Corrected] Layout/ArgumentAlignment: Use one level of indentation for arguments following the first line of a multi-line method call. (https://rubystyle.guide#no-double-indent)
    :second => 2
    ^^^^^^^^^^^^
spec/example.rb:7:5: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
    :second => 2
    ^^^^^^^^^^^^
spec/example.rb:7:12: C: [Corrected] Layout/ArgumentAlignment: Use one level of indentation for arguments following the first line of a multi-line method call. (https://rubystyle.guide#no-double-indent)
           :second => 2
           ^^^^^^^^^^^^
spec/example.rb:7:12: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
           :second => 2
           ^^^^^^^^^^^^
spec/example.rb:13:24: C: [Corrected] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
build(:factory_name, **options.merge( ...
                       ^^^^^^^^^^^^^^
spec/example.rb:19:24: C: [Corrected] Style/KeywordArgumentsMerging: Provide additional arguments directly rather than using merge. (https://rubystyle.guide#merging-keyword-arguments)
build(:factory_name, **options.merge(:first => 1, :second => 2))
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 7 offenses detected, 7 offenses corrected

1 error occurred:                                                                                                                                                                                                                                                                                  
Infinite loop detected in /Users/kevinrobell/Documents/source/.../spec/example.rb and caused by Layout/ArgumentAlignment, Layout/HashAlignment -> Layout/ArgumentAlignment, Layout/HashAlignment
configuration from /Users/kevinrobell/Documents/source/.../.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.                                                                                                                                                                    
https://github.com/rubocop/rubocop/issues                                                                                                                                                                                                                                                          
                                                                                                                                                                                                                                                                                                   
Mention the following information in the issue report:                                                                                                                                                                                                                                             
1.80.2 (using Parser 3.3.9.0, rubocop-ast 1.47.1, analyzing as Ruby 3.1, running on ruby 3.1.6) [arm64-darwin23]                                                                                                                                                                                   
Finished in 0.6055299999998169 second
```

Autocorrected example:
```rb
# frozen_string_literal: true

# 1. Infinite loop
build(
  :factory_name,
  **options, :first => 1,
    :second => 2
)

# 2. This works fine
build(:factory_name, **options, :first => 1,
  :second => 2)

# 3. This works fine
build(:factory_name, **options, :first => 1, :second => 2)
```

## RuboCop version

```
$ bundle exec rubocop -V
1.80.2 (using Parser 3.3.9.0, rubocop-ast 1.47.1, analyzing as Ruby 3.1, running on ruby 3.1.6) [arm64-darwin23]
  - rubocop-performance 1.26.0
  - rubocop-rails 2.33.4
  - rubocop-rspec 3.7.0
  - rubocop-factory_bot 2.27.1
  - rubocop-rspec_rails 2.31.0
```
